### PR TITLE
add SSL server_name logger

### PR DESCRIPTION
**Pull request by joeshaw**
This is the Server Name Indication (SNI) hostname provided in the TLS
handshake.

### CODE REVIEW

File: t/50access-log.t [hunk 1]
```diff
@@ -171,23 +171,23 @@ subtest 'extensions' => sub {
         sub {
             my $server = shift;
             sleep 1; # ensure check_port's SYN_ACK is delivered to the server before that generated by curl
-            system("curl --silent http://127.0.0.1:$server->{port}/ > /dev/null");
-            system("curl --silent --insecure @{[curl_supports_http2() ? ' --http1.1' : '']} https://127.0.0.1:$server->{tls_port}/ > /dev/null");
+            system("curl --silent http://localhost:$server->{port}/ > /dev/null");
```
Comment (joeshaw) [inline t/50access-log.t:174]: This change was necessary to get `curl` to send a value for SNI.  I considered using `--resolve` for this to keep the explicit binding of localhost to 127.0.0.1, but `nghttp` doesn't have an analog.

I can do this differently if there's a better way.

Comment (deweerdt) [reply t/50access-log.t:174]: @joeshaw a lot of tests rely on xip.io, `127.0.0.1.xip.io` should work.

Comment (deweerdt) [reply t/50access-log.t:174]: Ah, nm, i was distracted. `localhost` works.

**Comment by kazuho** [comment]
Thank you for the PR and for the accompanying tests! Merged.

**Pull request merged**